pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -182,6 +183,24 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 		if cosmosdb.IsErrorStatusCode(err, http.StatusPreconditionFailed) {
 			return nil, f.validateOpenShiftUniqueKey(ctx, doc)
 		}
+
```

Comment:
I don't know the logic here, but this will start billing from the point when the document is created, not cluster started. So in case of failure or long provisioning - it is running. Again, not sure if this is ok or not :man_shrugging:

Reply:
Yea, billing service will take care of that. It will query VMs running via CRP.
@shrutir25 to confirm

Reply:
We can move this to the backend as well but we definitely want to write the record before deploying the template. This is to ensure that we start billing for any VMs present in the cluster even if the template deployment fails because of a different error or cluster creation fails. The billing service will be able to find any running VMs once a record is present and if no VMs are presnt, it will be a no-op.